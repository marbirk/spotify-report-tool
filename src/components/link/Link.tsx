import React from 'react'
import ExternalLink from '../externalLink/ExternalLink'
import { Link as GatsbyLink } from 'gatsby'

interface LinkProps {
    children: React.ReactNode
    to: string
    className?: string
}

const Link = ({ children, to, ...other }: LinkProps) => {
    const internal = /^\/(?!\/)/.test(to)

    if (internal) {
        return (
            <GatsbyLink to={to} {...other}>
                {children}
            </GatsbyLink>
        )
    }

    return (
        <ExternalLink href={to} {...other}>
            {children}
        </ExternalLink>
    )
}

export default Link
