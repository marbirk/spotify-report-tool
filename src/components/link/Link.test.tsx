import React from 'react'
import { render, screen } from '@testing-library/react'

import Link from './Link'

describe('Link', () => {
    test('renders correctly gatsby link', () => {
        render(<Link to="/network">Internal Link</Link>)
        const component = screen.getByText('Internal Link')
        expect(component.getAttribute('target')).toBe(null)
        expect(component).toMatchSnapshot()
    })
    test('renders correctly external link', () => {
        render(<Link to={'https://test-url.de'}>External Link</Link>)
        const component = screen.getByText('External Link')
        expect(component.getAttribute('target')).toBe('_blank')
        expect(component).toMatchSnapshot()
    })
})
