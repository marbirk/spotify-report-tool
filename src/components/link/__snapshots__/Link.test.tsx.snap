// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly external link 1`] = `
<a
  href="https://test-url.de"
  rel="noreferrer noopener"
  target="_blank"
>
  External Link
</a>
`;

exports[`Link renders correctly gatsby link 1`] = `
<a
  href="/network"
>
  Internal Link
</a>
`;
